pubsub/src/lib.rs

```diff
@@ -5,9 +5,100 @@
 //! * [About Cloud Pub/Sub](https://cloud.google.com/pubsub/)
 //! * [Pub/Sub API Documentation](https://cloud.google.com/pubsub/docs)
 //!
-//! ## Quick Start
+//! ## Quickstart
 //!
-//! You can use [google-cloud-default](https://crates.io/crates/google-cloud-default) to create `ClientConfig`
+//! ### Authentication
+//!
+//! When you are not using an emulator you'll need to be authenticated.
+//! There are two ways to do that:
+//!
+//! #### Automatically
+//! You can use [google-cloud-default](https://crates.io/crates/google-cloud-default) to create [ClientConfig][crate::client::ClientConfig]
+//!
+//! This will try and read the credentials from a file specified in the environment variable `GOOGLE_APPLICATION_CREDENTIALS`, `GOOGLE_APPLICATION_CREDENTIALS_JSON` or
+//! from a metadata server.
+//!
+//! This is also described in [google-cloud-auth](https://github.com/yoshidan/google-cloud-rust/blob/main/foundation/auth/README.md)
+//!
+//! See [implementation](https://docs.rs/google-cloud-auth/0.9.1/src/google_cloud_auth/token.rs.html#59-74)
+//!
+//! #### Manually
+//!
+//! When you cant use the `gcloud` authentication but you have a different way to get your credentials (e.g a different environment variable)
+//! you can parse your own version of the 'credentials-file' and use it like that:
+//!
+//! ```
+//! # use google_cloud_auth::{credentials::CredentialsFile, project, token::DefaultTokenSourceProvider};
```

Comment:
I found it a bit difficult as Quickstart. If you need this kind of use case, I think it would be better to create `with_credentials(cred: CredentialsFile)` in google-cloud-default and use it.

Reply:
Thats a good idea!
I created the function `with_credentials` and adapted all tests and docs